app/Support/Modules/DatabaseActivator.php

```diff
@@ -0,0 +1,157 @@
+<?php
+
+namespace App\Support\Modules;
+
+use Illuminate\Config\Repository as Config;
+use Illuminate\Container\Container;
+use Illuminate\Filesystem\Filesystem;
+use Nwidart\Modules\Contracts\ActivatorInterface;
+use Nwidart\Modules\Module;
+
+class DatabaseActivator implements ActivatorInterface
+{
+    /**
+     * Laravel config instance
+     *
+     * @var Config
+     */
+    private $config;
+
+    /**
+     * @var Filesystem
+     */
+    private $files;
+
+    /**
+     * The module path.
+     *
+     * @var string|null
+     */
+    protected $path;
+
+    /**
+     * The scanned paths.
+     *
+     * @var array
+     */
+    protected $paths = [];
+
+    /**
+     * Array of modules activation statuses
+     *
+     * @var array
+     */
+    private $modulesStatuses;
+
+    public function __construct(Container $app, $path = null)
+    {
+        $this->config = $app['config'];
+        $this->files = $app['files'];
+        $this->modulesStatuses = $this->getModulesStatuses();
+        $this->path = $path;
+    }
+
+    /**
+     * Get modules statuses, from the database
+     *
+     * @return array
+     */
+    private function getModulesStatuses(): array
+    {
+        $modules = \App\Models\Module::all();
+        $retVal = [];
+        foreach ($modules as $i) {
+            $retVal[$i->name] = $i->enabled;
+        }
+        return $retVal;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function reset(): void
+    {
+        (new \App\Models\Module())->truncate();
+        $this->modulesStatuses = [];
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function enable(Module $module): void
+    {
+        $this->setActiveByName($module->getName(), true);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function disable(Module $module): void
+    {
+        $this->setActiveByName($module->getName(), false);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function hasStatus(Module $module, bool $status): bool
+    {
+        if (!isset($this->modulesStatuses[$module->getName()])) {
```

Comment:
Use the model directly here. Also, update the inline doc here to specify the type of `Module` that's being passed in